### PR TITLE
Feat/split lucene to admin and core

### PR DESCRIPTION
**Pull request by bkapustik**
# Motivation

Adds support for administration/live site managmant only

## Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

## How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: .azuredevops/pipelines/build-and-release.yaml [hunk 1]
```diff
@@ -47,73 +47,51 @@ stages:
...
-          - name: ProjectFolder
-            value: src/Kentico.Xperience.Lucene
-
-          - name: ProjectFilePath
-            value: src/Kentico.Xperience.Lucene/Kentico.Xperience.Lucene.csproj
+          - name: slnPath
+            value: src/Kentico.Xperience.Lucene.Libs.sln
 
         steps:
-          - task: PowerShell@2
```
Comment (michalJakubis) [inline .azuredevops/pipelines/build-and-release.yaml:57]: I would rather leave the Node setup steps here, we need Node.js to build Admin/Client.

File: .azuredevops/pipelines/build-and-release.yaml [hunk 2]
```diff
@@ -47,73 +47,51 @@ stages:
...
+            command: build
+            projects: ${{ variables.slnPath }}
+            configuration: ${{ variables.Configuration }}
+            arguments: --no-restore
+          env:
+            AuthenticodeClientSecret: $(AuthenticodeClientSecret)
+            # Roll-forward behavior set for AzureSignTool dotnet tool (see .config\dotnet-tools.json) which requires .Net 6.0 runtime
+            DOTNET_ROLL_FORWARD: Major
+
+        - template: steps.yml
```
Comment (michalJakubis) [inline .azuredevops/pipelines/build-and-release.yaml:82]: This template will not work unless you implement the steps.yml configuration. https://github.com/Kentico/xperience-by-kentico-crm/blob/main/.azuredevops/pipelines/steps.yml

File: src/Kentico.Xperience.Lucene/LuceneSearchModule.cs
Comment (michalJakubis) [inline src/Kentico.Xperience.Lucene/LuceneSearchModule.cs:1]: This package should ideally wrap both Core and Admin packages, so we avoid duplicated code across the solution. Would it be possible to just reference `Kentico.Xperience.Lucene.Admin` in this project and reuse all the functionality from there?

File: examples/DancingGoat/appsettings.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
   },
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "CMSConnectionString": "Data Source=GIGACHAD;Initial Catalog=xperience-lucene;Integrated Security=True;Persist Security Info=False;Connect Timeout=60;Encrypt=False;Current Language=English;"
+    "CMSConnectionString": ""
   },
-  "CMSHashStringSalt": "sal"
+  "CMSHashStringSalt": "sal",
+  "WebCrawlerBaseUrl": ""
```
Comment (michalJakubis) [inline examples/DancingGoat/appsettings.json:20]: Is this setting necessary for successful setup of the example project? Do we have any documentation covering this setting?

**Review by michalJakubis — CHANGES_REQUESTED** [review]

File: examples/DancingGoat/appsettings.json (hunk 3 above)
Comment (bkapustik) [reply examples/DancingGoat/appsettings.json:20]: It is necessary for advanced examples. I will add documentation for this.

File: src/Kentico.Xperience.Lucene/LuceneSearchModule.cs
Comment (bkapustik) [reply src/Kentico.Xperience.Lucene/LuceneSearchModule.cs:1]: Yes, I have changed it

**Review by michalJakubis — APPROVED** [review]

**Pull request merged**